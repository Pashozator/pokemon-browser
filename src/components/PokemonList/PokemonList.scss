@import '../../styles/media-queries';

.pokemon-list {
  align-items: center;
  justify-content: center;
  flex-direction: column;
  display: flex;

  @include for-up-sm() {
    align-items: center;
    justify-content: flex-start;
    flex-wrap: wrap;
    flex-direction: row;
  }
}
