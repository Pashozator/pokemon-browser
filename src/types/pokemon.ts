export interface Pokemon {
	id: string;
	number: number;
	name: string;
	image: string;
	maxHP: number;
	fleeRate: number;
	maxCP: number;
	classification: string;
	types: string[];
	resistant: string[];
	weaknesses: string[];
	evolutions: Pokemon[];
}
