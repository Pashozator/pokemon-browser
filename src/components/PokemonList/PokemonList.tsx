import React from 'react';
import './PokemonList.scss';
import { Pokemon } from '../../types/pokemon';
import SkeletonCardLoading from '../SkeletonCardLoading/SkeletonCardLoading';
import PokemonCard from '../PokemonCard/PokemonCard';

interface PokemonListProps {
	pokemons: Pokemon[];
	loading?: boolean;
	className?: string;
}

const PokemonList: React.FC<PokemonListProps> = ({ pokemons = [], loading = false, className = '' }) => {
	return (
		<section className={`pokemon-list ${className}`}>
			{loading && (
				<>
					<SkeletonCardLoading/>
					<SkeletonCardLoading/>
					<SkeletonCardLoading/>
					<SkeletonCardLoading/>
				</>
			)}
			{pokemons.map(pokemon => <PokemonCard key={pokemon.id} pokemon={pokemon}/>)}
		</section>
	);
};

export default PokemonList;
