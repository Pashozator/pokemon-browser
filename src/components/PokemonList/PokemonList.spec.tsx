import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Pokemon } from '../../types/pokemon';
import PokemonList from './PokemonList';

describe('PokemonList component', () => {
	it('should render', () => {
		const pokemons: Pokemon[] = [{
			id: 'test',
			name: 'Pikachu',
			image: 'image'
		}];

		const wrapper = shallow(<PokemonList pokemons={pokemons}/>);

		expect(wrapper.exists()).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
