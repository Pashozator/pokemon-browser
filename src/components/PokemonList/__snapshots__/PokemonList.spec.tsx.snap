// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList component should render 1`] = `
<section
  className="pokemon-list "
>
  <PokemonCard
    key="test"
    pokemon={
      Object {
        "id": "test",
        "image": "image",
        "name": "Pikachu",
      }
    }
  />
</section>
`;
